src/BlockPatterns.php

```diff
@@ -58,6 +69,12 @@ public function register_block_patterns() {
 			return;
 		}
 
+		$supports_wc_blocks_patterns = get_theme_support( 'woocommerce-blocks-patterns' );
+
+		if ( ! $supports_wc_blocks_patterns ) {
+			return;
+		}
```

Comment:
I would recommend adding a dedicated filter for this purpose as well (our equivalent of the core's `should_load_remote_block_patterns` filter):

```suggestion
		/**
		 * Filter to disable WooCommerce block patterns.
		 *
		 * @since $:VID:$
		 *
		 * @param bool $should_load_block_patterns
		 */
		$should_load_block_patterns = apply_filters( 'should_load_woocommerce_block_patterns', true );

		if ( ! $should_load_block_patterns || ! $supports_wc_blocks_patterns ) {
			return;
		}
```

Reply:
🤔 The `should_load_remote_block_patterns` filter in WP core is only for affecting _remote_ pattern loading. These aren't remotely loaded patterns in Woo Blocks. The [WP core implementation of the filter](https://github.com/WordPress/wordpress-develop/blob/ca6a093af2e966dd8f2b5c183a5699ff786bec8b/src/wp-includes/block-patterns.php#L205-L214), references both the theme_support check _and_ the remote filter. However, only the theme supports check is used for [what is bundled with WP core](https://github.com/WordPress/wordpress-develop/blob/ca6a093af2e966dd8f2b5c183a5699ff786bec8b/src/wp-includes/block-patterns.php#L20). 

IMO I don't see much need for including a filter.

Reply:
> The [WP core implementation of the filter](https://github.com/WordPress/wordpress-develop/blob/ca6a093af2e966dd8f2b5c183a5699ff786bec8b/src/wp-includes/block-patterns.php#L205-L214), references both the theme_support check and the remote filter. However, only the theme supports check is used for [what is bundled with WP core](https://github.com/WordPress/wordpress-develop/blob/ca6a093af2e966dd8f2b5c183a5699ff786bec8b/src/wp-includes/block-patterns.php#L20).

That's true; our patterns are not remotely loaded: the main reason I suggested this change is because, IMO, adding a dedicated filter is more intuitive to any developer digging our codebase than relying on `remove_theme_support`, which is documented as a function the should be used within `functions.php` and generally used to override support from the parent theme, which can be confusing considering WooCommerce Blocks is a plugin and not a theme.

Reply:
I understand the points you are making, however, I disagree it's more (or even less) intuitive to use a filter especially given there is already a precedent set with how to programatically stop the patterns included in WP core from being registered. If anything, I think the proposed documentation here could be updated to remove the reference to a theme's function.php file since really all that is needed is for the `remove_theme_support` call to be invoked via a callback on the `after_theme_setup` action (which could be in a theme's `functions.php` file or a plugin).

Reply:
> I disagree it's more (or even less) intuitive to use a filter especially given there is already a precedent set with how to programatically stop the patterns included in WP core from being registered. If anything, I think the proposed documentation here could be updated to remove the reference to a theme's function.php file since really all that is needed is for the remove_theme_support call to be invoked via a callback on the after_theme_setup action (which could be in a theme's functions.php file or a plugin).

Gotcha, I see your point & it makes sense. IMO more than just the documentation, what doesn't quite fit the criteria is also the core's nomenclature for these `*_theme_support` functions, with the assumption that themes are the only ones registering and de-registering their support for a particular feature when, in reality, plugins also can (and are!).

TBH, I'm still not a big fan of relying on those functions in their current shape and form for all the reasons mentioned earlier, but using those to be aligned with the core's implementation is also valid, so I'll defer to you :).

Adding to my to-do list here to open a PR on Core to update those docblocks & also suggest introducing functions with revised nomenclature, less theme-oriented for these purposes.